.github/actions/build-push-docker-gcr/action.yml

```diff
@@ -0,0 +1,85 @@
+name: Build and push images for given branch
+description: Builds the Zeebe Docker image with tag following SemVer and push it gcr.io/
+
+inputs:
+  branch:
+    description: 'Specifies the branch, for which the image should be build'
+    default: 'main'
+    required: false
+  secret_vault_address:
+    description: 'secret vault url'
+    required: true
+  secret_vault_roleId:
+    description: 'secret vault roleId'
+    required: true
+  secret_vault_secretId:
+    description: 'secret valut secret id'
+    required: true
+
+outputs:
+  image:
+    description: "Fully qualified image name"
+    value: ${{ steps.build-docker.outputs.image }}
+
+runs:
+  using: composite
+  steps:
+    # Dynamic environment variables are not supported by GHA
+    # https://brandur.org/fragments/github-actions-env-vars-in-env-vars
+    #
+    # Since we run the workflow either on demand or via schedule we need to assign some defaults
+    # Furthermore we have branches like stable/1.0 where we have to replace certain patterns, in order to use the branch name as docker image tag
+    - id: evaluate-inputs
+      name: Evaluate Inputs
+      shell: bash
+      run: |
+        branch=${BRANCH/\//-}
+        branch=${branch//\./-}
+        branch=${branch:-main}
+        echo "BRANCH_NAME=$branch" >> $GITHUB_ENV
+      env:
+        BRANCH: "${{ inputs.branch }}"
+    # We need to check out the evaluated branch and setup java (incl. maven), so we can retrieve the current project version
+    # The version is necessary, since CC Saas only accepts SemVer for docker image tags (need to start with a version tag)
+    - uses: actions/checkout@v3
+      with:
+        ref: "${{ github.event.inputs.branch }}"
+    - uses: actions/setup-java@v3.6.0
```

Comment:
:x: `setup-zeebe` will set up Java. We can already call it here before we even generate the TAG. So, e.g. the ordering can be:

```yaml
   - uses: actions/checkout@v3
   - uses: hashicorp/vault-action@v2.4.2
   - uses: ./.github/actions/setup-zeebe
   - uses: ./.github/actions/build-zeebe
   - id: generate-tag
   - uses: docker/login-action@v2
   - uses: ./.github/actions/build-docker
```